File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/AllPrimitiveLeafTypes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.houseelectrics.serializer.test
{
    public class AllPrimitiveLeafTypes
    {
        public AllPrimitiveLeafTypes()
        {
            ALong = 123;
            ALongRef = 234;
            AShort = 124;
            ALongRef = 456;
            AChar = 'c';
            AString2 = "hi";
            AUint32 = 3232;
            AUint64 = 6464;
        }

        public long aLong = 123;
        public short aShort = 124;
        public long? aLongRef = 456;
        public string aString, aString2 = "hi";
        public char aChar = 'c';

        public UInt32 aUint32 = 3232;
        public UInt64 aUint64 = 6464;


        public long ALong { get; set; }
        public short AShort { get; set; }
        public long ALongRef { get; set; }
        public long? ALongRefNullable { get; set; }
        public string AString { get; set; }
        public string AString2 { get; set; }
        public char AChar { get; set; }

        public UInt32 AUint32 { get; set; }
        public UInt64 AUint64 { get; set; }

        public static String[] testFieldExpressions = { "aLong", "aShort", "aLongRef", "aString", "aChar", "aString2", "aUint32", "aUint64" };
        public static Object[] testExpectedFieldValues(AllPrimitiveLeafTypes template)
        {
            return new Object[] { template.aLong, template.aShort, template.aLongRef, template.aString, template.aChar.ToString(), template.aString2, template.aUint32, template.aUint64 };
        }
        public static String[] testPropertyExpressions = { "ALong", "AShort", "ALongRef", "ALongRefNullable", "AString", "AChar", "AString2", "AUint32", "AUint64" };
        public static Object[] testExpectedPropertyValues(AllPrimitiveLeafTypes template)
        {
            return new Object[] { template.ALong, template.AShort, template.ALongRef, template.ALongRefNullable, template.AString, template.AChar/*.ToString()*/, template.AString2, template.AUint32, template.AUint64 };
        }


    }
}

File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/AllPrimitiveLeafTypes.java
package com.houseelectrics.serializer.test;

/**
 * Created by roberttodd on 30/11/2014.
 */
public class AllPrimitiveLeafTypes
{
    public AllPrimitiveLeafTypes()
    {
        setALong(123);
        setALongRef(234l);
        setAInt(324);
        setAIntRef(345);
        setAShort((short)124);
        setAChar('c');
        setAString("hi");
        setAByte((byte)127);
    }

    public long aLong = 23;
    public Long aLongRef = 789l;
    public int aInt = 463;
    public Integer aIntRef = 247;
    public short aShort = 124;
    public char aChar = 'c';
    public String aString, aString2 = "hi";
    public byte aByte=126;

    private long _aLong = 23;
    private Long _aLongRef = 789l;
    private int _aInt = 463;
    private Integer _aIntRef = 247;
    private short _aShort = 124;
    private char _aChar = 'c';
    private String _aString = "hi";
    private byte _aByte;



    public static String[] testFieldExpressions = { "aLong", "aLongRef", "aInt", "aIntRef", "aShort", "aChar", "aString", "aString2", "aByte" };
    public static Object[] testExpectedFieldValues(AllPrimitiveLeafTypes template)
    {
        return new Object[] { template.aLong, template.aLongRef, template.aInt, template.aIntRef, template.aShort, template.aChar, template.aString, template.aString2, template.aByte };
    }
    public static String[] testPropertyExpressions = { "ALong", "ALongRef", "AInt", "AIntRef", "AShort", "AChar", "AString", "AByte" };
    public static Object[] testExpectedPropertyValues(AllPrimitiveLeafTypes template)
    {
        return new Object[] { template.getALong(), template.getALongRef(), template.getAInt(), template.getAIntRef(),
                 template.getAShort(), template.getAChar(), template.getAString(), template.getAByte() };
    }


    public long getALong() {return _aLong;}
    public void setALong(long value) { this._aLong = value; }

    public Long getALongRef()
    {
        return _aLongRef;
    }

    public void setALongRef(Long _aLongRef)
    {
        this._aLongRef = _aLongRef;
    }

    public int getAInt()
    {
        return _aInt;
    }

    public void setAInt(int _aInt)
    {
        this._aInt = _aInt;
    }

    public Integer getAIntRef()
    {
        return _aIntRef;
    }

    public void setAIntRef(Integer _aIntRef)
    {
        this._aIntRef = _aIntRef;
    }

    public short getAShort()
    {
        return _aShort;
    }

    public void setAShort(short _aShort)
    {
        this._aShort = _aShort;
    }

    public char getAChar()
    {
        return _aChar;
    }

    public void setAChar(char _aChar)
    {
        this._aChar = _aChar;
    }

    public String getAString()
    {
        return _aString;
    }

    public void setAString(String _aString)
    {
        this._aString = _aString;
    }

    public void setAByte(byte value) {this._aByte = value;}
    public byte getAByte() {return _aByte;}

}
